### PR TITLE
Add Pulse Connect Secure

### LINKED ISSUE #1927: Add Pulse Secure Connect integration support
Hi,

There was a need to integrate with pulse secure connect. What is needed to create an integration for this solution in elastic solution.
I am ready to provide any necessary information to do so.

Best regards

### PR DESCRIPTION
**Pull request by legoguy1000**
## What does this PR do?

Add Pulse Connect Secure integration

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/master/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] If I'm introducing a new feature, I have modified the Kibana version constraint in my package's `manifest.yml` file to point to the latest Elastic stack release (e.g. `^7.13.0`).

## Author's Checklist

- [ ]

## Related issues
- Closes #1927

### CODE REVIEW

**Comment by legoguy1000** [comment]
I started this and let it drop. I'll try to finish it up this week/weekend so it can be reviewed and merged.

**Comment by zez3** [comment]
Great work, I was hoping to get this in 7.16
Is this really expecting kibana v8 to function?
I saw some option like bello on other packages:
kibana.version: "^7.16 | ^8.0.0"

**Comment by legoguy1000** [comment]
Thats up to @P1llus and team.  Every integration i've made recently has been for 8.0.0. I can change it to 7.16 if they're good with it.

**Comment by P1llus** [comment]
@jamiehynds ?

**Comment by legoguy1000** [comment]
Ready for initial review. Passed all local tests.

**Comment by jamiehynds** [comment]
> Thats up to @P1llus and team. Every integration i've made recently has been for 8.0.0. I can change it to 7.16 if they're good with it.

Given that it will take some time for folks to upgrade to 8.0, if there's no technical blocker to including the integration as part of 7.16, fine on my end.

**Comment by P1llus** [comment]
Feel free to bump this to 1.0.0 since we have proper system and pipeline test coverage @legoguy1000

File: packages/pulse_connect_secure/data_stream/log/elasticsearch/ingest_pipeline/default.yml [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+  - set:
+      field: event.kind
+      value: event
+  - set:
+      field: event.category
+      value: network
+  - grok:
+      field: message
+      patterns:
+        - 'Agent login %{WORD:_tmp.outcome} for %{DATA}%{SESSION} from %{IP} with %{GREEDYDATA:user_agent.original}.'
```
Comment (P1llus) [inline packages/pulse_connect_secure/data_stream/log/elasticsearch/ingest_pipeline/default.yml:59]: Does this mean we only support these audit log events, are there other events we are missing?
If we only support a subset of logs we should maybe add that to the docs somewhere?

File: packages/pulse_connect_secure/data_stream/log/elasticsearch/ingest_pipeline/default.yml [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+      ignore_failure: true
+      ignore_missing: true
+  - set:
+      field: event.outcome
+      value: failure
+      if: '["failed"].contains(ctx._tmp?.outcome)'
+  - set:
+      field: event.outcome
+      value: success
+      if: '["successful", "succeeded"].contains(ctx._tmp?.outcome)'
```
Comment (P1llus) [inline packages/pulse_connect_secure/data_stream/log/elasticsearch/ingest_pipeline/default.yml:82]: I think these will fail if outcome can ever be null or not exist? Might want to add a != null check first.

File: packages/pulse_connect_secure/manifest.yml [hunk 3]
```diff
@@ -0,0 +1,29 @@
+name: pulse_connect_secure
+title: Pulse Connect Secure
+version: 0.0.1
```
Comment (P1llus) [inline packages/pulse_connect_secure/manifest.yml:3]: Feel free to bump to 1.0.0 and change the release tag :)

**Review by P1llus — APPROVED** [review]

File: packages/pulse_connect_secure/data_stream/log/elasticsearch/ingest_pipeline/default.yml (hunk 1 above)
Comment (legoguy1000) [reply packages/pulse_connect_secure/data_stream/log/elasticsearch/ingest_pipeline/default.yml:59]: No but I have limited sample logs and limited info on what the log messages mean to decode them.  That's kinda why I want to leave it at experimental.

File: packages/pulse_connect_secure/manifest.yml (hunk 3 above)
Comment (legoguy1000) [reply packages/pulse_connect_secure/manifest.yml:3]: see above comment

File: packages/pulse_connect_secure/data_stream/log/elasticsearch/ingest_pipeline/default.yml (hunk 2 above)
Comment (legoguy1000) [reply packages/pulse_connect_secure/data_stream/log/elasticsearch/ingest_pipeline/default.yml:82]: I'll update this.

File: packages/pulse_connect_secure/data_stream/log/elasticsearch/ingest_pipeline/default.yml (hunk 1 above)
Comment (zez3) [reply packages/pulse_connect_secure/data_stream/log/elasticsearch/ingest_pipeline/default.yml:59]: @legoguy1000 
I can help with logs if needed. We have the older 8.6 as well as the new 9.x version in prod

Comment (olmanolsztyn) [reply packages/pulse_connect_secure/data_stream/log/elasticsearch/ingest_pipeline/default.yml:59]: I can help with logs too, but in logs im interested in there is the same pattern.

**Pull request merged**